crates/compiler/gen_dev/src/generic64/storage.rs

```diff
@@ -759,17 +798,53 @@ impl<
                 debug_assert!(from_offset % 8 == 0);
                 debug_assert!(size % 8 == 0);
                 debug_assert_eq!(size, layout_interner.stack_size(*layout));
-                self.with_tmp_general_reg(buf, |_storage_manager, buf, reg| {
-                    for i in (0..size as i32).step_by(8) {
-                        ASM::mov_reg64_base32(buf, reg, from_offset + i);
-                        ASM::mov_base32_reg64(buf, to_offset + i, reg);
-                    }
-                });
+                self.copy_symbol_to_stack_offset_help(buf, size, from_offset, to_offset)
             }
             x => todo!("copying data to the stack with layout, {:?}", x),
         }
     }
 
+    pub fn copy_symbol_to_stack_offset_help(
+        &mut self,
+        buf: &mut Vec<'a, u8>,
+        size: u32,
+        from_offset: i32,
+        to_offset: i32,
+    ) {
+        let mut copied = 0;
+        let size = size as i32;
+
+        self.with_tmp_general_reg(buf, |_storage_manager, buf, reg| {
+            for _ in (0..(size - copied)).step_by(8) {
```

Comment:
I think this is wrong.

Lets say we want to copy 4 bytes. This first for loop will run once because `(0..4).step_by(8)` produces `[0]`. So this would emit an 8 byte copy. I think you need to wrap every for loop in a conditional that you have at lest step_by bytes.

Reply:
yes, fixed